python/lsst/sims/catUtils/mixins/PhotometryMixin.py

```diff
@@ -620,3 +621,133 @@ def get_lsst_magnitudes(self):
+            # it's "dry run" to determine what columns are required from
+            # the database
+            return numpy.zeros((len(bandpassDict.keys()),0))
+
+        magListOut = []
+
+        for sedName, magNorm in zip(sedNameList, magNormList):
+            magTag = bandpassTag+'_'+sedName
+            if sedName not in self._ssmMagNormDict or magTag not in self._ssmMagDict:
+                dummySed = Sed()
+                dummySed.readSED_flambda(os.path.join(self._file_dir, self._spec_map[sedName]))
+                fnorm = dummySed.calcFluxNorm(magNorm, self._normalizing_bandpass)
+                dummySed.multiplyFluxNorm(fnorm)
+                magList = bandpassDict.magListForSed(dummySed)
+                self._ssmMagDict[magTag] = magList
+                self._ssmMagNormDict[sedName] = magNorm
+            else:
+                dmag = magNorm - self._ssmMagNormDict[sedName]
+                magList = self._ssmMagDict[magTag] + dmag
+            magListOut.append(magList)
+
+        return numpy.array(magListOut).transpose()
+
+
+    @compound('lsst_u','lsst_g','lsst_r','lsst_i','lsst_z','lsst_y')
+    def get_lsst_magnitudes(self):
+        """
+        getter for LSST magnitudes of solar system objects
+        """
+
+        if not hasattr(self, 'lsstBandpassDict'):
+            self.lsstBandpassDict = BandpassDict.loadTotalBandpassesFromFiles()
+
+        return self._magnitudeGetter(self.lsstBandpassDict)
+
+
+    @compound('dmagTrailing', 'dmagDetection')
+    def get_ssm_dmag(self):
+        """
+        This getter will calculate:
+
+        dmagTrailing: the offset in m5 used to represent the loss in signal to noise
+        resulting from the fact that the object's motion smears out its PSF
+
+        dmagDetection: the offset in m5 used to represent the shift in detection
+        threshold resulting from the fact that the object's motion smears out
+        its PSF
+        """
+
+        if self.obs_metadata.seeing is None:
+            raise RuntimeError("Cannot calculate dmagTraling/dmagDetection. "
+                               "Your catalog's ObservationMetaData does not "
+                               "specify seeing.")
+
+        if not isinstance(self.obs_metadata.bandpass, str):
+            raise RuntimeError("dmagTrailing/dmagDetection calculation is confused. "
+                               "Your catalog's ObservationMetaData contains multiple "
+                               "bandpasses.  It is unclear what seeing value you mean. "
+                               "Re-create your catalog with only one bandpass.")
+
+        if not hasattr(self, 'photParams') or self.photParams is None:
+            raise RuntimeError("You cannot calculate dmagTrailing/dmagDetection. "
+                               "Your catalog does not have an associated PhotometricParameters "
+                               "member variable.  It is impossible to know what the exposure time is.")
```

Comment:
Hmm. Seems like it might be handy to either be able to change the seeing for PhotometricParameters or have exposure time available in obsMetadata (although I'd suggest implementing this as : a real default value is in photometricParameters and the obsMetadata *may* have an overriding value set -- this means that sims_photUtils still doesn't have to use obsMetadata, but that if someone is running through a series of pointings from an opsim run, they can easily change the visit exposure time for each exposure via the same interface they are updating the rest of the obs metadata).